uiowa/uiowa
# UIowa Hours

Resolves #4344 

## ToDo

- [x] Require `uiowa_hours.settings.group` config be set to place block. (@joewhitsitt)
- [x] Ignore `uiowa_hours.settings`configuration. (@joewhitsitt)
- [x] Make render better (card wrap, fanciness?) - _meeting with Ben to iron this out_
- [x] Status indicator js (@richardbporter)
- [x] ajax layer in drupal announce and other a11y helpers (@joewhitsitt)
- [x] check for "hours" category? (@richardbporter)
- [x] better no results - separate between error and "closed" (@richardbporter)
- [x] default render today for pick a date option
- [x] can we assume one result per resource for a day? the key says otherwise (@richardbporter)
- [x] look through code to make sure everything is named suitably (what is this thing? what will it become?)





# How to test

- `ddev blt sync --site recserv.uiowa.edu`
- Enable hours split and import config.
- Attempt to place block on page, discover that you must select a resource group first before placing a block.
- Go and select a resource group (recserv fusion). Save.
- Place a hours block on a page. Play with the block config options to make sure they work as expected.
- Try the date filter (keyboard, datepicker), after selecting a date, inspect source and find #drupal-live-announce to see that it announces the change to the user.
- ??
- Profit.
